### ISSUE TITLE
PA_TASK_ID, PA_TASK_NAME, PA_JOB_ID, PA_JOB_NAME not set for use in bash task

### ISSUE THREAD

**Issue by sandrineBeauche** [ow2-proactive/scheduling#2296]
I tried to launch this workflow on the scheduler:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<job
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns="urn:proactive:jobdescriptor:3.2"
     xsi:schemaLocation="urn:proactive:jobdescriptor:3.2 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.2/schedulerjob.xsd"
    name="bash test" 
    priority="normal"
    cancelJobOnError="false">
  <description>
    <![CDATA[ A workflow that executes bash in JVM. ]]>
  </description>
  <taskFlow>
    <task name="Bash_Task">
      <description>
        <![CDATA[ The simplest task, ran by a bash engine. ]]>
      </description>
      <scriptExecutable>
        <script>
          <code language="bash">
            <![CDATA[
echo $PA_TASK_ID
echo $PA_TASK_NAME
echo $PA_JOB_ID
echo $PA_JOB_NAME
]]>
          </code>
        </script>
      </scriptExecutable>
      <controlFlow block="none"></controlFlow>
    </task>
  </taskFlow>
</job>
```

This workflow should print the value of the PA_TASK_ID, PA_TASK_NAME, PA_JOB_ID, PA_JOB_NAME variables. Instead there is no output.

**Comment by sandrineBeauche** [comment]
I submit this workflow too:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<job
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xmlns="urn:proactive:jobdescriptor:3.2"
     xsi:schemaLocation="urn:proactive:jobdescriptor:3.2 http://www.activeeon.com/public_content/schemas/proactive/jobdescriptor/3.2/schedulerjob.xsd"
    name="echo variables" 
    priority="normal"
    cancelJobOnError="false">
  <description>
    <![CDATA[ A workflow that executes groovy in JVM. ]]>
  </description>
  <genericInformation>
    <info name="var1" value="val1"/>
    <info name="var2" value="val2"/>
  </genericInformation>
  <taskFlow>
    <task name="Groovy_Task">
      <description>
        <![CDATA[ The simplest task, ran by a groovy engine. ]]>
      </description>
      <scriptExecutable>
        <script>
          <code language="groovy">
            <![CDATA[
println("task_id_prop="+System.getProperty("pas.task.id"));
println("job_id_env="+System.getenv("PAS_TASK_ID"));
]]>
          </code>
        </script>
      </scriptExecutable>
      <controlFlow block="none"></controlFlow>
    </task>
  </taskFlow>
</job>
```

This workflow try to get the task id firstly by java property and then by environment variable.
Here is the output:

```
[1780000@62-210-124-102.rev.poneytelecom.eu;10:47:25] task_id_prop=1780000 
[1780000@62-210-124-102.rev.poneytelecom.eu;10:47:25] task_id_env=null 
```

**Comment by lpellegr** [comment]
@sandrineBeauche What is the version/commit of the scheduler used? I ask because if I remember correctly, some changes regarding variables (standardization) have been introduced with the _Speed branch_.

If you use the master, then for accessing ProActive system variables from a native task (e.g. Bash) you need to use the following notation:

```bash
echo $variables_PA_TASK_ID
```

From a script task (e.g. Groovy), you can retrieve the value through the _variables_ map:

```groovy
println variables.get("PA_TASK_ID")
```

If you use a bash task, the script engine will delegate the execution to the native shell. Thus, you can see it as a native task and the first example must be used for retrieving the task ID.

It is [half-explained](http://jenkins.activeeon.com/job/documentation/11/artifact/build/asciidoc/html5/user/ProActiveUserGuide.html#_proactive_system_variables) in the documentation and [some examples are given](http://jenkins.activeeon.com/job/documentation/11/artifact/build/asciidoc/html5/user/ProActiveUserGuide.html#_variables_quick_reference).

Besides, the documentation states that _${}_ syntax can be used in the **workflow XML definition depending of the context**. 

In conclusion, it seems not to be a bug but the manner to access variables which has changed since version _6.1.0_.

**Comment by sandrineBeauche** [comment]
This used version was 6.0.2. The syntax you gave me works well with 6.3. So I can close this issue now.
